deploy/crds/metal3.io_baremetalhosts_crd.yaml

```diff
@@ -74,6 +74,9 @@ spec:
                   description: The name of the secret containing the BMC credentials
                     (requires keys "username" and "password").
                   type: string
+                insecure:
```

Comment:
The name "insecure" implies more to me than skipping the CA verification. I like Zane's proposal to call it disableHostVerification (or skipHostVerification or disable/skipCAVerification) better because it's more explicit about what's being turned off.

Reply:
I actually prefer 'insecure' because it is extremely insecure. I mean, it's not as insecure as it could be I guess, but that's not really how security works. It should feel a bit icky every time you have to set this flag.
That said, I wouldn't block using a more descriptive name. Searching around, I think disableCertificateVerification is probably the most accurate term for what's happening here.

Reply:
Yeah, my main concern was the implied scope. Does "insecure" also mean "log passwords to stdout"? Maybe that's not something I should be worried about, though.

Reply:
Ah yeah, I see what you mean. _In the context of TLS options_ 'insecure' is a perfectly adequate name, but there's currently nothing to give that context.
If we were to move this inside a struct called something like `TLSOptions` then I'd advocate for keeping `Insecure`. This would make sense if we were also going to e.g. implement Dmitry's suggestion of adding an option for a custom CA cert, and any other potential options (the only one that comes to mind is client certificates, but I assume there are no BMCs that use those).
If we don't do that then I agree that we should use a name that doesn't require any context to interpret.